js/PixelPainter.js

```diff
@@ -0,0 +1,313 @@
+        }
+        pixelMatrix.push(collumArray);
+      }
+    };
+    makeCanvasMatrix(100, 66);
+
+    const makeColorSwath = function(x, y) {
+      let colorCollumCount = 1;
+      let colorBoxCount = 1;
+
+      for (let i = 0; i < x; i++) {
+        let colorCollum = document.createElement('div');
+        colorCollum.id = 'colorBox' + colorCollumCount;
+        colorArea.appendChild(colorCollum);
+        colorCollumCount++;
+        for (let a = 0; a < y; a++) {
+          let colorBox = document.createElement('div');
+          colorBox.id = 'color' + colorBoxCount;
+          colorBox.className = 'colorBoxes';
+          colorBox.style.backgroundColor = '#' + (((1 << 24) * Math.random()) | 0).toString(16);
+          colorBoxCount++;
+          colorCollum.appendChild(colorBox);
+        }
+      }
+    };
+    makeColorSwath(4, 16);
+
+    function randomColorGenerator() {
+      let getColorBox = document.getElementsByClassName('colorBoxes');
+      for (let i = 0; i < getColorBox.length; i++) {
+        getColorBox[i].style.backgroundColor = '#' + (((1 << 24) * Math.random()) | 0).toString(16);
+      }
+    }
+
+    let currentColor;
+    let isMouseDown = false;
+
+    const getColorClass = document.getElementsByClassName('colorBoxes');
+    for (var i = 0; i < getColorClass.length; i++) {
+      getColorClass[i].addEventListener('click', getColor);
+    }
+
+    function getColor() {
+      currentColor = this.style.backgroundColor;
+    }
+
+    let getAllBoxes = document.getElementsByClassName('boxes');
+    for (var i = 0; i < getAllBoxes.length; i++) {
+      getAllBoxes[i].addEventListener('click', setColor);
+      getAllBoxes[i].addEventListener('mousedown', getReadyToDrag);
+      getAllBoxes[i].addEventListener('mouseover', startDragging);
+      getAllBoxes[i].addEventListener('mouseup', stopDraggingMe);
+    }
+
+    function setColor() {
+      this.style.backgroundColor = currentColor;
+    }
+
+    function getReadyToDrag() {
+      isMouseDown = true;
+    }
+
+    function startDragging() {
+      if (isMouseDown == true) {
+        this.style.backgroundColor = currentColor;
+      }
+    }
+
+    function stopDraggingMe() {
+      isMouseDown = false;
+    }
+
+    const getClearButton = document.getElementById('clearButton');
+    getClearButton.addEventListener('click', clearCanvas);
+
+    function clearCanvas() {
+      for (var i = 0; i < getAllBoxes.length; i++) {
+        getAllBoxes[i].style.backgroundColor = 'FFFFFF';
+      }
+    }
+
+    const getEraseButton = document.getElementById('eraseButtonon');
+    getEraseButton.addEventListener('click', startErase);
+
+    function startErase() {
+      if (currentColor !== 'FFFFFF') {
+        getEraseButton.style.backgroundColor = 'FFFFFF';
+      }
+      currentColor = 'FFFFFF';
+    }
+
+    const getNewColorButt = document.getElementById('newColors');
+    getNewColorButt.addEventListener('click', randomColorGenerator);
+  }
+  if (y.matches) {
```

Comment:
So you still have a 

`if (x.matches)`

and a 

`if (y.matches)`

But the two conditionals are largely the same. It's better to find the differences between your x matching and your y matching and try to set those differences to variables. Then your matchMyMedia function can start by using an if to set those variables, then you can continue with building out the rest of your pixel painter without having so much duplicate code.